From cf32454f2ff255ba1c98fdd1cf4c4cee77cb6700 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster <raster@rasterman.com>
Date: Wed, 29 Oct 2008 21:28:59 +1100
Subject: [PATCH] exquisite: Fix the package so it actually can be a splash screen app

This fixes the exquisitie packages so they provide an initscript and all the
appropriate sysvinit goop in the .bb file so exquisite can be a splash app
if its installed and enabled by the initscripts.
---
 packages/e17/exquisite/exquisite-init |   40 +++++++++++++++++++++++++++++++++
 packages/e17/exquisite_svn.bb         |   26 ++++++++++++++++++++-
 2 files changed, 65 insertions(+), 1 deletions(-)
 create mode 100644 packages/e17/exquisite/exquisite-init

diff --git a/packages/e17/exquisite/exquisite-init b/packages/e17/exquisite/exquisite-init
new file mode 100644
index 0000000..6c94fdc
--- /dev/null
+++ b/packages/e17/exquisite/exquisite-init
@@ -0,0 +1,40 @@
+#!/bin/sh
+
+for x in $(cat /proc/cmdline); do
+  case $x in
+    splash=false)
+      echo "Splash Disabled"
+      exit 0;
+      ;;
+  esac
+done
+
+. /etc/init.d/functions
+
+### Default config here - gets overriden later
+THEME="-t default"
+FPS="-fps 30"
+### Override config - if there is a config file - load it.
+if test -f /etc/exquisite/config; then
+  . /etc/exquisite/config
+fi
+
+args=''
+case `machine_id` in
+  "hp_ipaq_h3100"|"hp_ipaq_h3800")
+     args='-rot 90' ;;
+  "hp_ipaq_3600"|"hp_ipaq_3700"|"hp_ipaq_h3900"|"htc_universal"|*collie|*poodle|*akita|*spitz|*borzoi)
+     args='-rot 270' ;;
+esac
+args=$args" $FPS $THEME"
+
+export EXDIR=/mnt/.exquisite
+export EXQUISITE_IPC=$EXDIR/exquisite
+mount tmpfs -t tmpfs $EXDIR -o,size=40k
+exquisite -fb -fs $args &
+
+exquisite-write -wait 20
+exquisite-write "PULSATE"
+
+# kill off exquisite in 180 seconds if its not gone yet
+(sleep 180; exquisite-write "QUIT")&
diff --git a/packages/e17/exquisite_svn.bb b/packages/e17/exquisite_svn.bb
index 4777226..f1c8908 100644
--- a/packages/e17/exquisite_svn.bb
+++ b/packages/e17/exquisite_svn.bb
@@ -2,8 +2,32 @@ DESCRIPTION = "A psplash replacement for display"
 LICENSE = "MIT BSD"
 DEPENDS = "eet evas ecore embryo edje"
 PV = "0.0.1+svnr${SRCREV}"
-PR = "r1"
+PR = "r2"
+RDEPENDS = "initscripts"
+RRECOMMENDS_${PN} = "exquisite-themes"
+
+SRCNAME = "exquisite"
 
 inherit e
 
 EXTRA_OECONF = "--with-edje-cc=${STAGING_BINDIR_NATIVE}/edje_cc"
+
+SRC_URI += "file://exquisite-init"
+
+inherit update-rc.d
+
+do_install_prepend() {
+  install -d ${D}/mnt/.exquisite/
+  install -d ${D}${sysconfdir}/init.d/
+  install -m 0755 ${WORKDIR}/exquisite-init ${D}${sysconfdir}/init.d/exquisite
+}
+
+
+do_install_append() {
+  rm -rf ${D}${datadir}/exquisite/data/fonts/*
+}
+
+INITSCRIPT_NAME = "exquisite"
+INITSCRIPT_PARAMS = "start 01 S . stop 20 0 1 6 ."
+
+FILES_${PN} += "/mnt/.exquisite/"
-- 
1.5.4.3

